### PR TITLE
feat(API): Add TriggerPolicy to pipeline_spec

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: api/v2alpha1/pipeline_spec.proto [hunk 1]
```diff
@@ -367,6 +367,21 @@ message PipelineTaskSpec {
   // Reference to a component.  Use this field to define either a DAG or an
   // executor.
   ComponentRef component_ref = 7;
+
+  // Trigger policy defines how the task gets triggered. If a task is not
+  // triggered, it will run into SKIPPED state.
```
Comment (hongye-sun) [inline api/v2alpha1/pipeline_spec.proto:372]: I forgot to fix the comment here. We changed the state to be NOT_TRIGGERED to be backward compatible. SKIPPED state is still preserved for caching purpose. Could you change the state here to NOT_TRIGGERED and i will send another CL to fix internal proto comments.

**Review by hongye-sun — CHANGES_REQUESTED** [review]

File: api/v2alpha1/pipeline_spec.proto (hunk 1 above)
Comment (chensun) [reply api/v2alpha1/pipeline_spec.proto]: Done.

**Review by hongye-sun — COMMENTED** [review]
/lgtm

**Pull request merged**